### PR TITLE
Comments fix

### PR DESCRIPTION
**Pull request by VolodymyrTrach**

### CODE REVIEW

File: modules/comment/comment.service.js [hunk 1]
```diff
@@ -10,22 +13,30 @@ class CommentsService {
   async getAllCommentsByProduct(id) {
     const product = await Product.findById(id);
     if (!product) {
-      throw new Error(PRODUCT_NOT_FOUND);
+      throw new Error(COMMENT_NOT_FOUND);
     }
     return Comments.find({ product: id });
   }
 
-  updateComment(id, comments) {
```
[the remark this answers is not in the document]
Comment (VolodymyrTrach) [reply modules/comment/comment.service.js:18]: done

File: modules/comment/comment.service.js [hunk 2]
```diff
@@ -10,22 +13,30 @@ class CommentsService {
...
+    return Comments.findByIdAndUpdate(id, comment, { new: true });
   }
 
-  addComment(data) {
+  async addComment(id, data) {
+    const product = await Product.findById(id);
+    if (!product) {
+      throw new Error(COMMENT_FOR_NOT_EXISTING_PRODUCT);
+    }
     const comments = new Comments(data);
```
[the remark this answers is not in the document]
Comment (VolodymyrTrach) [reply modules/comment/comment.service.js]: done

File: modules/comment/comment.service.js [hunk 3]
```diff
@@ -1,31 +1,48 @@
...
+      throw new Error(COMMENT_NOT_FOUND);
     }
-    return Comments.find({ product: id });
+    return Comment.find({ product: id });
   }
 
-  updateComment(id, comments) {
-    return Comments.findByIdAndUpdate(id, comments, { new: true });
+  async updateComment(id, comment) {
+    const commentToUpdate = await Comment.findByIdAndUpdate(id, comment, {
```
Comment (Alan18081) [inline modules/comment/comment.service.js:22]: Please, rename variable to the `updatedComment`

File: modules/comment/comment.service.js [hunk 4]
```diff
@@ -1,31 +1,48 @@
...
+      throw new Error(COMMENT_FOR_NOT_EXISTING_PRODUCT);
+    }
+    const comment = new Comment(data);
+    return comment.save();
   }
 
-  deleteComment(id) {
-    return Comments.findByIdAndDelete(id);
+  async deleteComment(id) {
+    const commentToDelete = await Comment.findByIdAndDelete(id);
```
Comment (Alan18081) [inline modules/comment/comment.service.js:41]: Please, rename it to the `deletedComment`

File: modules/comment/comment.service.js (hunk 3 above)
Comment (Alan18081) [reply modules/comment/comment.service.js:22]: And convert if statement to the `if (!updatedComment) { throw new Error(COMMENT_NOT_FOUND) }`

File: modules/comment/comment.service.js [hunk 5]
```diff
@@ -1,31 +1,48 @@
...
+    if (commentToDelete) {
```
Comment (Alan18081) [inline modules/comment/comment.service.js:42]: And convert if statement to the `if (!deleteComment) { throw new Error(COMMENT_NOT_FOUND) }`

File: modules/comment/comment.service.js (hunk 3 above)
Comment (VolodymyrTrach) [reply modules/comment/comment.service.js:22]: renamed and converted

File: modules/comment/comment.service.js (hunk 4 above)
Comment (VolodymyrTrach) [reply modules/comment/comment.service.js:41]: renamed

File: modules/comment/comment.service.js (hunk 5 above)
Comment (VolodymyrTrach) [reply modules/comment/comment.service.js:42]: converted

**Review by Alan18081 — APPROVED** [review]

**Pull request merged**